### ISSUE TITLE
Pattern Assembler - Custom Home template core/group validation error

### ISSUE THREAD

**Issue by miksansegundo** [Automattic/wp-calypso#73001]
There are unexpected unsaved changes in the editor when landing on it at the end pattern assembler flow. That's why the "save" button is enabled sometimes.

This issue doesn't happen for all patterns and doesn't happen only for Header patterns. It could be failing because of having nested Group blocks, that issue was reported in the past. See p1616561793255800/1616555061.247200-slack-CS0SC36SX

In the following screenshot, the Home template conatins only one section pattern, the save button is enabled, and the validation error for the `core/group` is in the console.

<img width="1280" alt="Screenshot 2566-02-06 at 18 48 51" src="https://user-images.githubusercontent.com/1881481/216964082-1e76440b-0478-4fea-8e3e-d974ec7b8413.png">

The issue seems related to the validation of the block `core/group`, which is used in the template `Home`. Note that the validation error has emptied the part `Content retrieved from post body:`, which is unexpected. 

The error could be triggered by one of the `core/group` [deprecations](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/group/deprecated.js), but that doesn't seem to be the case.

**Comment by dsas** [comment]
I think I've found the problem

`verticalize_template` receives a template_id (e.g. bc3/home) and template_content (i.e. the pattern html). It's supposed to **insert the template_content inside the "main" block** of the template_id, however what it is actually doing is **replacing the template_content of the "main" block** of the template_id. This happens at fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Siregvpnyvmr%2Qfgnegre%2Qpbagrag.cuc%3Se%3Q6sn59r0o%23117-og. 

This means the HTML - the `<main>` opening and closing tags, isn't in the post body, but as the gutenberg comment tag is there, in javascript the group blocks `save` function is triggered, and it can regenerate what the tag should've been. 

As an example, the template that template_id refers to looks like
```
(
    [1] => Array
        (
            [blockName] => core/group
            [attrs] => Array
                (
                    [tagName] => main
                )

            [innerBlocks] => Array
                (
                )

            [innerHTML] =>
                <main class="wp-block-group">
                </main>

            [innerContent] => Array
                (
                    [0] =>
                <main class="wp-block-group">
                </main>

                )

        )

)
```

After inserting the content you expect:

```
(
    [1] => Array
        (
            [blockName] => core/group
            [attrs] => Array
                (
                    [tagName] => main
                )

            [innerBlocks] => Array
                (
                )

            [innerHTML] =>
                <main class="wp-block-group">
                    <!-- wp:group -->
                    <div class="wp-block-group">
                        <!-- wp:paragraph --><p>Foobarbaz</p><!-- /wp:paragraph -->
                    </div>
                    <!-- /wp:group -->
                </main>

            [innerContent] => Array
                (
                    [0] =>
                <main class="wp-block-group">
                    <!-- wp:group -->
                    <div class="wp-block-group">
                        <!-- wp:paragraph --><p>Foobarbaz</p><!-- /wp:paragraph -->
                    </div>
                    <!-- /wp:group -->
                </main>

                )

        )

)
```

but what you get is:

```
(
    [1] => Array
        (
            [blockName] => core/group
            [attrs] => Array
                (
                    [tagName] => main
                )

            [innerBlocks] => Array
                (
                )

            [innerHTML] =>
                    <!-- wp:group -->
                    <div class="wp-block-group">
                        <!-- wp:paragraph --><p>Foobarbaz</p><!-- /wp:paragraph -->
                    </div>
                    <!-- /wp:group -->

            [innerContent] => Array
                (
                    [0] =>
                    <!-- wp:group -->
                    <div class="wp-block-group">
                        <!-- wp:paragraph --><p>Foobarbaz</p><!-- /wp:paragraph -->
                    </div>
                    <!-- /wp:group -->

                )

        )

)
```

**Comment by arthur791004** [comment]
Proposing a fix: D100528-code

**Comment by arthur791004** [comment]
Closing it via D100528-code

**Comment by miksansegundo** [comment]
Great teamwork! Thank you @dsas for identifying the bug and @arthur791004 for the fix! 🙌

The fix D100528-code solves the validation error and therefore the editor doesn't enable the save button.

One of the motivations to fix this issue is that the LaunchPad save modal shows unexpectedly when the editor loads at the end of the assembler flow. That [issue](https://github.com/Automattic/dotcom-forge/issues/1592) is still reproducible when we choose a header or footer pattern with navigation.
